### DISCUSSION TITLE
Possible? Get latest version of 2 packages have a common dependency

### DISCUSSION THREAD

**Discussion by IAXES** [renovatebot/renovate#20678] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, select which platform you are using.

GitHub Enterprise Server

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I have a project, `breakfast`, that depends on two modules: `spam`, and `eggs`.  Both of these two modules depend on a common/shared dependency, `food`. In order for `breakfast` to work, it must use versions of `spam` and `eggs` that were both built against the exact same version of `food` (i.e. maybe it's a communications library that `spam` and `eggs` use to somehow interact w/ each other).

Is it possible, for example (e.g. w/ custom managers and data sources) to have a rule in Renovate that equates to:

- Update the `breakfast project` to ...
- Use version `^1.2.3` of spam, and version `^4.5.6` of eggs ...
- Such that both of these modules were built against the same version of `food`? (i.e. key on complex criteria).

I could approximate this with with synchronized builds, but those will still be subject to race conditions. The `spam` and `eggs` projects are maintained by separate teams, so they can't be combined into a single repo. For now, there's a somewhat convoluted shell script we have in our CI pipelines to accomplish this, but I'm curious if we could accomplish this in a "cleaner" way w/ Renovate.

Thank you.

### What you tried so far.

N/A. Trying to determine if this is possible and/or practical w/ Renovate.

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment, marked as answer]
I'm not aware of any way to do this in Renovate. You could naturally group `spam` and `eggs` updates into a single branch/PR, but that doesn't solve the challenge that you need Renovate to then resolve and compare transitive dependencies

**Comment by viceice** [comment]
not yet possible

**Comment by IAXES** [comment]
Got it. Based on the docs, it seemed unlikely, but thanks for the clarification (also thanks for writing/maintaining this awesome tool!).
